sdk-core/src/main/java/care/data4life/sdk/RecordService.kt

```diff
@@ -0,0 +1,1146 @@
+import care.data4life.sdk.config.DataRestriction.DATA_SIZE_MAX_BYTES
+import care.data4life.sdk.config.DataRestrictionException
+import care.data4life.sdk.lang.CoreRuntimeException
+import care.data4life.sdk.lang.D4LException
+import care.data4life.sdk.lang.DataValidationException
+import care.data4life.sdk.model.CreateResult
+import care.data4life.sdk.model.DeleteResult
+import care.data4life.sdk.model.DownloadResult
+import care.data4life.sdk.model.DownloadType
+import care.data4life.sdk.model.FetchResult
+import care.data4life.sdk.model.Meta
+import care.data4life.sdk.model.ModelVersion
+import care.data4life.sdk.model.Record
+import care.data4life.sdk.model.AppDataRecord
+import care.data4life.sdk.model.EmptyRecord
+import care.data4life.sdk.model.UpdateResult
+import care.data4life.sdk.model.definitions.BaseRecord
+import care.data4life.sdk.model.definitions.DataRecord
+import care.data4life.sdk.network.model.DecryptedRecord
+import care.data4life.sdk.network.model.DecryptedAppDataRecord
+import care.data4life.sdk.network.model.EncryptedKey
+import care.data4life.sdk.network.model.EncryptedRecord
+import care.data4life.sdk.network.model.definitions.DecryptedBaseRecord
+import care.data4life.sdk.network.model.definitions.DecryptedDataRecord
+import care.data4life.sdk.network.model.definitions.DecryptedFhirRecord
+import care.data4life.sdk.util.Base64.decode
+import care.data4life.sdk.util.Base64.encodeToString
+import care.data4life.sdk.util.HashUtil.sha1
+import care.data4life.sdk.util.MimeType
+import care.data4life.sdk.util.MimeType.Companion.recognizeMimeType
+import io.reactivex.Completable
+import io.reactivex.Observable
+import io.reactivex.Single
+import org.threeten.bp.LocalDate
+import org.threeten.bp.LocalDateTime
+import org.threeten.bp.ZoneId
+import org.threeten.bp.format.DateTimeFormatter
+import org.threeten.bp.format.DateTimeFormatterBuilder
+import java.io.IOException
+import java.util.*
+
+internal class RecordService(
+        private val partnerId: String,
+        private val alias: String,
+        private val apiService: ApiService,
+        private val tagEncryptionService: TagEncryptionService,
+        private val taggingService: TaggingService,
+        private val fhirService: FhirService,
+        private val attachmentService: AttachmentService,
+        private val cryptoService: CryptoService,
+        private val errorHandler: SdkContract.ErrorHandler
+) {
+    internal enum class UploadDownloadOperation {
+        UPLOAD, DOWNLOAD, UPDATE
+    }
+
+    internal enum class RemoveRestoreOperation {
+        REMOVE, RESTORE
+    }
+
+    @Throws(DataRestrictionException.UnsupportedFileType::class,
+            DataRestrictionException.MaxDataSizeViolation::class)
+    @JvmOverloads
+    fun <T : DomainResource> createRecord(
+            resource: T,
+            userId: String,
+            annotations: List<String> = listOf()
+    ): Single<Record<T>> {
+        checkDataRestrictions(resource)
```

Comment:
We need to have data restrictions for AppData too, as the payload size that could be processed by our API RecordService is limited to 1MB (current limit to be checked with PHDP team).

Reply:
So you mean the byte resource is restricted to 1 MB?

Reply:
we could do this as separate ticket